libraries/Adafruit_LittleFS/src/Adafruit_LittleFS_File.cpp

```diff
@@ -130,151 +152,260 @@ bool File::open (char const *filepath, uint8_t mode)
-  return ch;
+  int ret = -1;
+  this->DoNotCallFromOutsideClass_LockFilesystem();
+  if (!this->_is_dir)
+  {
+    uint32_t pos = lfs_file_tell(_fs->getFS(), _file);
+    uint8_t ch = 0;
+    if (lfs_file_read(_fs->getFS(), _file, &ch, 1) > 0)
+    {
+      ret = static_cast<int>(ch);
+    }
+    (void)lfs_file_seek(_fs->getFS(), _file, pos, LFS_SEEK_SET);
+  }
+  this->DoNotCallFromOutsideClass_UnlockFilesystem();
+  return ret;
 }
 
 int File::available (void)
 {
-  return size() - position();
+  int ret = 0;
+  this->DoNotCallFromOutsideClass_LockFilesystem();
+  if (!this->_is_dir)
+  {
+    uint32_t size = lfs_file_size(_fs->getFS(), _file);
+    uint32_t pos  = lfs_file_tell(_fs->getFS(), _file);
+    ret = size - pos;
+  }
+  this->DoNotCallFromOutsideClass_UnlockFilesystem();
+  return ret;
 }
 
 bool File::seek (uint32_t pos)
 {
-  VERIFY(!_is_dir, false);
-  return lfs_file_seek(_fs->getFS(), _file, pos, LFS_SEEK_SET) >= 0;
+  bool ret = false;
+  this->DoNotCallFromOutsideClass_LockFilesystem();
+  if (!this->_is_dir)
+  {
+    ret = lfs_file_seek(_fs->getFS(), _file, pos, LFS_SEEK_SET) >= 0;
+  }
+  this->DoNotCallFromOutsideClass_UnlockFilesystem();
+  return ret;
 }
 
 uint32_t File::position (void)
 {
-  VERIFY(!_is_dir, 0);
-  return lfs_file_tell(_fs->getFS(), _file);
+  uint32_t ret = 0;
+  this->DoNotCallFromOutsideClass_LockFilesystem();
+  if (!this->_is_dir)
+  {
+    ret = lfs_file_tell(_fs->getFS(), _file);
+  }
+  this->DoNotCallFromOutsideClass_UnlockFilesystem();
+  return ret;
 }
 
 uint32_t File::size (void)
 {
-  VERIFY(!_is_dir, 0);
-  return lfs_file_size(_fs->getFS(), _file);
+  uint32_t ret = 0;
+  this->DoNotCallFromOutsideClass_LockFilesystem();
+  if (!this->_is_dir)
+  {
+    ret = lfs_file_size(_fs->getFS(), _file);
+  }
+  this->DoNotCallFromOutsideClass_UnlockFilesystem();
+  return ret;
+
 }
 
 void File::flush (void)
 {
-  VERIFY(!_is_dir,);
-  lfs_file_sync(_fs->getFS(), _file);
+  this->DoNotCallFromOutsideClass_LockFilesystem();
+  if (!this->_is_dir)
+  {
+    lfs_file_sync(_fs->getFS(), _file);
+  }
+  this->DoNotCallFromOutsideClass_UnlockFilesystem();
+  return;
 }
 
 void File::close (void)
 {
-  if ( (*this) )
+  this->DoNotCallFromOutsideClass_LockFilesystem();
+  this->_close();
+  this->DoNotCallFromOutsideClass_UnlockFilesystem();
+}
+void File::_close(void)
+{
+  if ( this->_isOpen() )
   {
-    if ( _is_dir )
+    if ( this->_is_dir )
     {
       lfs_dir_close(_fs->getFS(), _dir);
       rtos_free(_dir);
       _dir = NULL;
 
-      if ( _dir_path ) rtos_free(_dir_path);
+      if ( this->_dir_path ) rtos_free(_dir_path);
       _dir_path = NULL;
     }
     else
     {
-      lfs_file_close(_fs->getFS(), _file);
+      lfs_file_close(this->_fs->getFS(), _file);
       rtos_free(_file);
       _file = NULL;
     }
   }
 }
 
+[[deprecated("Recommend use of isOpen() for clarity")]]
 File::operator bool (void)
+{
+  bool ret = false;
+  this->DoNotCallFromOutsideClass_LockFilesystem();
```

Comment:
there is no need to lock fs for this function. We can also make bool() as inline verion of isOpen()

Reply:
OK.